packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/utils/textBoxMapping/MappingSourceTextProvider.ts

```diff
@@ -0,0 +1,61 @@
+import { TextSpan } from '../../types';
+import { TextProvider } from './TextProvider';
+import { TextNormalizer } from '../common/TextNormalizer';
+import minBy from 'lodash/minBy';
+import { spanGetText, spanLen, START } from '../common/textSpanUtils';
+import { TextLayoutCell } from '../textLayout/types';
+
+const debugOut = require('debug')?.('pdf:mapping:MappingSourceTextProvider');
+function debug(...args: any) {
+  debugOut?.apply(null, args);
+}
+
+export class MappingSourceTextProvider {
```

Comment:
Reminder: the README does a good job of detailing the overall process. But each of these files should have a short description comment block as well.